feat(#2611): add segmented variant to tabs

vanessatran-ddi opened the pull request:
# Before (the change)

 - Tabs component only supported the default underline-style variant
  - Individual tabs could not be disabled - all tabs were always selectable
  - No support for skipping tabs during keyboard navigation
  - No `variant` prop on `<GoabTabs>`
  - No `disabled` prop on `<GoabTab>`
  
# After (the change)
- Added `variant` prop to `<GoabTabs>` with two options:
    - `"default"` - Standard underline-style tabs (existing behavior)
    - `"segmented"` - New pill-style tabs with animated sliding background indicator
  - Added `disabled` prop to `<GoabTab>`:
    - Disabled tabs cannot be clicked or selected
    - Disabled tabs are skipped during keyboard navigation (Arrow keys, Home, End)
    - Disabled tabs display with grayed-out styling and `cursor: not-allowed`
    - If `initialTab` points to a disabled tab, the first enabled tab is automatically selected
  - Added accessibility attributes for disabled tabs (`aria-disabled="true"`, `tabindex="-1"`)
  - Segmented variant uses MutationObserver and ResizeObserver to resize the pill when tab content changes (e.g., badge count updates)
  
<img width="756" height="278" alt="image" src="https://github.com/user-attachments/assets/42869050-55e7-472e-a0c7-30137b1a0233" />

## Make sure that you've checked the boxes below before you submit the PR

- [x] I have read and followed the [setup steps](https://goa-dio.atlassian.net/wiki/spaces/DS/pages/3086385489/Contribution+-+web+content#%5BinlineExtension%5DCode-contributions)
- [x] I have created necessary unit tests
- [x] I have tested the functionality in both React and Angular.

## Steps needed to test
 ### React
  1. Run `npx nx serve prs-react`
  2. Navigate to `/features/2611` to test:
     - Segmented variant vs default variant comparison
     - Long labels handling
     - Disabled tabs in both variants
     - Edge case: initialTab pointing to disabled tab
  3. Navigate to `/features/2611-tabs-disabled` to verify:
     - When `initialTab=1` and Tab 1 is disabled, Tab 2 is automatically selected

  ### Angular
  1. Run `npx nx serve prs-angular`
  2. Navigate to `/features/2611` to test the same scenarios as React
  3. Navigate to `/features/2611-tabs-disabled` to verify the disabled tab edge case

willcodeforcoffee reviewed (dismissed):
Looks good! Great demos, too! Really impressive. 😃

ArakTaiRoth commented on libs/web-components/src/components/tabs/Tabs.svelte:
```diff
@@ -152,7 +220,9 @@
   function handleHashChange() {
     const tabIndexFromHash = getTabIndexFromHash();
```
So, the hash can be changed to a disabled tab's hash, this doesn't activate the disabled tab (good), but it does change the URL to include the hash of the disabled tab (bad)

This can be seen in the Test Case 3 scenario I added to the `tabs-disabled` pr files.

Do we want to have it revert the hash shown in the URL to the last good activated hash? @chrisolsen

vanessatran-ddi commented on libs/web-components/src/components/tabs/Tabs.svelte:
```diff
@@ -230,6 +321,33 @@
+
+      _segmentedIndicatorLeft = selectedRect.left - tabsRect.left - 1;
+      _segmentedIndicatorWidth = selectedRect.width;
+      _segmentedIndicatorHeight = selectedRect.height;
```
@chrisolsen 
the sliding animation needs js because the indicator has to know where to move and how is the width. Each tab has different width depending on text length and badge (or other react node), if I use only css, it won't have the animation.


https://github.com/user-attachments/assets/eae91d35-cff2-4424-8f27-df8beb3a12e1

vanessatran-ddi commented:
> @vanessatran-ddi Looks good overall, great work.
> 
> The only additional thing I noticed:
> 
> * on the v1 version, the colour of the text for both the active tab and the inactive tab is too light. Lets adjust it to greyscale-800 on the inactive tab, and text-default on the active tab.
> 
> ### Current:
> <img alt="Image" width="955" height="135" src="https://private-user-images.githubusercontent.com/82968683/532452939-c781daa9-66bd-40d6-ab68-4d614160a3a9.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc5MTAyMjgsIm5iZiI6MTc2NzkwOTkyOCwicGF0aCI6Ii84Mjk2ODY4My81MzI0NTI5MzktYzc4MWRhYTktNjZiZC00MGQ2LWFiNjgtNGQ2MTQxNjBhM2E5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTA4VDIyMDUyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBlYmMyZmE2NDA0OWM5ZmE1OTBiYTU5NzliOWIxODNlZGEyZTQ5MjcwMDJlYmU1MmE3Mzg0MjNjYTI0NTNiYjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.uMhBJ0FN2EgDpEbuPf5DScBBwk0PL9vKH5cqfXN9kHs">
> ### Update to:
> <img alt="Image" width="955" height="135" src="https://private-user-images.githubusercontent.com/82968683/532452455-7f049242-2022-42c4-a121-de7eb93f8c63.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc5MTAyMjgsIm5iZiI6MTc2NzkwOTkyOCwicGF0aCI6Ii84Mjk2ODY4My81MzI0NTI0NTUtN2YwNDkyNDItMjAyMi00MmM0LWExMjEtZGU3ZWI5M2Y4YzYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTA4VDIyMDUyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkM2RhM2Y5ODAzNjBkMzgyN2JmMDZlZWU3MzVmN2E5M2MyZjcxYmEyMzcwOWQ4YmZiZWEyNzNhY2UxNGU4MmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.y6pOU70qOHWGq5gHKjowvq-kNeUltNH6GhYMl9FxvhU">

Hi @twjeffery  I addressed your comment on [this commit ](https://github.com/GovAlta/ui-components/pull/3302/commits/59079b4f8cdd7c8a5ec0466a368086e78dd3ace9)

chrisolsen commented:
:tada: This PR is included in version 1.10.0-next.1 :tada:

The release is available on:
- [npm package (@next dist-tag)](https://www.npmjs.com/package/@abgov/ui-components-common/v/1.10.0-next.1)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/common-v1.10.0-next.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

chrisolsen commented:
:tada: This PR is included in version 4.10.0-next.1 :tada:

The release is available on:
- [npm package (@next dist-tag)](https://www.npmjs.com/package/@abgov/angular-components/v/4.10.0-next.1)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/angular-v4.10.0-next.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

chrisolsen commented:
:tada: This PR is included in version 1.40.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@abgov/web-components/v/1.40.0)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/web-components-v1.40.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

chrisolsen commented:
:tada: This PR is included in version 4.10.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@abgov/angular-components/v/4.10.0)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/angular-v4.10.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
